docs/proposals/admission_control_changes.md

```diff
@@ -0,0 +1,485 @@
+result in undesirable or incorrect behavior. For example, if you run
+`ResourceQuota` before `LimitRanger`, you could end up incorrectly calculating
+quota because it's possible for `LimitRanger` to modify pod values after
+`ResourceQuota` performed its quota checks.
+
+Because of this, we want to move to a model where in-tree admission plugin
+execution is strongly ordered and not under the administrator's control. We
+would therefore deprecate the `--admission-control` setting from
+`pkg/genericapiserver/options.ServerRunOptions`. During the deprecation period,
+if this flag is set, we will display a warning while continuing to honor the values
+specified by the flag (i.e., the administrator can override the hard-coded ordering).
+
+
+## One config file per plugin
+
+All admission plugins currently share a single config file, specified via
+`--admission-control-config-file`. This is problematic if multiple plugins
+use the same names for their configuration sections. For
+example, if plugin `Foo` wants
+
+```yaml
+myConfig:
+  someField: red
+```
+
+and plugin `Bar` wants
+
+```yaml
+myConfig:
+  someField:
+  - one
+  - two
+  - three
+```
+
+this won't work because of the naming collision and the type of `someField` is
+different for the two plugins. What we'd like to see instead is:
+
+1. deprecate `--admission-control-config-file`
+1. replace it with `--admission-plugin PLUGIN=CONFIG_FILE` (or an equivalent
+   setup in a structured configuration, once we've fully moved over)
+
+You would specify `--admission-plugin` once per plugin that you want to
+configure. An example of this would be
+
+```
+--admission-plugin AlwaysPullImages=/path/to/config.yaml --admission-plugin MyPlugin=/some/other/config.yaml
+```
+
+An example of a structured configuration for admission control inside an
+apiserver configuration file might look like this:
+
+```yaml
+admissionConfig:
+  AlwaysPullImages:
+    configurationFile: /path/to/config.yaml
+  MyPlugin:
+    configurationFile: /some/other/config.yaml
+  
+```
+
+## Ability to enable/disable individual plugins
+
+With a fixed list of plugins enabled by default, administrators will need to be
+able to override the defaults and disable certain plugins that are normally on,
+and enable plugins that are normally off. We are proposing that the admission
+control section in the apiserver configuration file include support for this. We
+can use `enabled` as seen below:
+
+```yaml
+admissionConfig:
+  AlwaysPullImages:
+    # disabled by default, so enable it
+    enabled: true
+    configurationFile: /path/to/config.yaml
+  MyPlugin:
+    # enabled by default, so disable it
+    enabled: false
+    configurationFile: /some/other/config.yaml
+```
+
+Note, the enabling/disabling mechanism described in this proposal only works
+when using a structured configuration setup for the apiserver. The
+`--admission-plugin` flag described above does not provide a means to enable or
+disable a plugin (TODO: should it????).
```

Comment:
I have to say, I'm liking my proposal of having them all in a directory and named sensibly :)

You could extend it to having the yaml file include a enable/disable line.